src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,66 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SPLIT_STRING = ",";
+    private static final String STRING_BUY = "buy";
+    private static final String STRING_SUPPLY = "supply";
+    private static final String STRING_RESULT = "result";
 
     public void getStatistic(String fromFileName, String toFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder = createReport(readingFomCsv(fromFileName));
+        writingToCsv(stringBuilder.toString(), toFileName);
+    }
+
+    private void writingToCsv(String stringData, String toFileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+            writer.write(stringData);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write file: ", e);
+        }
+    }
+
+    private String readingFomCsv(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
```

Comment:
Better use List<String> - you are going to split it anyway. So, why do you concat all lines to the one String (or SB)?